packages/expo-notifications/android/src/main/java/expo/modules/notifications/NotificationsModule.java

```diff
@@ -0,0 +1,337 @@
+    intervalSchedulerModel.setScheduledTime(System.currentTimeMillis() + ((Double) options.get("interval")).longValue());
+    intervalSchedulerModel.setInterval(((Double) options.get("interval")).longValue()); // on iOS we cannot change interval
+
+    SchedulerImpl scheduler = new SchedulerImpl(intervalSchedulerModel);
+
+    SchedulersManagerProxy.getInstance(mContext.getApplicationContext()).addScheduler(
+        scheduler,
+        (String id) -> {
+          if (id == null) {
+            promise.reject(new UnableToScheduleException());
+            return false;
+          }
+          promise.resolve(id);
+          return true;
+        }
+    );
+  }
+
+  @ExpoMethod
+  public void scheduleNotificationWithCalendar(HashMap data, final HashMap options, final Promise promise) {
+    data = new NotificationScoper(mModuleRegistry.getModule(StringScoper.class)).scope(data);
+    data.put(NOTIFICATION_APP_ID_KEY, mAppId);
+
+    HashMap<String, Object> details = new HashMap<>();
+    details.put("data", data);
+
+    Cron cron = createCronInstance(options);
+
+    CalendarSchedulerModel calendarSchedulerModel = new CalendarSchedulerModel();
+    calendarSchedulerModel.setappId(mAppId);
+    calendarSchedulerModel.setDetails(details);
+    calendarSchedulerModel.setRepeat(options.containsKey("repeat") && (Boolean) options.get("repeat"));
+    calendarSchedulerModel.setCalendarData(cron.asString());
+
+    SchedulerImpl scheduler = new SchedulerImpl(calendarSchedulerModel);
+
+    SchedulersManagerProxy.getInstance(mContext.getApplicationContext()).addScheduler(
+        scheduler,
+        (String id) -> {
+          if (id == null) {
+            promise.reject(new UnableToScheduleException());
+            return false;
+          }
+          promise.resolve(id);
+          return true;
+        }
+    );
+  }
+
+  public void onCreate(ModuleRegistry moduleRegistry) {
+    mModuleRegistry = moduleRegistry;
+
+    AppIdProvider appIdProvider = moduleRegistry.getModule(AppIdProvider.class);
+    mAppId = appIdProvider.getAppId();
+
+    createDefaultChannel();
+    mChannelManager = getChannelManager();
+
+    PostOfficeProxy.getInstance().registerModuleAndGetPendingDeliveries(mAppId, this);
+  }
+
+  private void createDefaultChannel() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      CharSequence name = NOTIFICATION_DEFAULT_CHANNEL_NAME;
+      int importance = NotificationManager.IMPORTANCE_DEFAULT;
+      NotificationChannel channel = new NotificationChannel(NOTIFICATION_DEFAULT_CHANNEL_ID, name, importance);
+      NotificationManager notificationManager = mContext.getSystemService(NotificationManager.class);
+      notificationManager.createNotificationChannel(channel);
+    }
+  }
+
+  protected ChannelManager getChannelManager() {
+    return ThreadSafeChannelManager.getInstance();
+  }
+
+  public void onDestory() {
```

Comment:
Would this code work better if it would read `onDestroy`? Pay attention to IDE clues! 😅

Reply:
oops